src/main/java/ch/njol/skript/expressions/ExprArmorSlot.java

```diff
@@ -18,69 +18,86 @@
  */
 package ch.njol.skript.expressions;
 
+import java.util.Arrays;
 import java.util.Locale;
+import java.util.stream.Stream;
 
 import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
 import org.bukkit.inventory.EntityEquipment;
 import org.eclipse.jdt.annotation.Nullable;
 
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.util.slot.EquipmentSlot;
 import ch.njol.skript.util.slot.EquipmentSlot.EquipSlot;
 import ch.njol.skript.util.slot.Slot;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Armour Slot")
-@Description("A part of a player's armour, i.e. the boots, leggings, chestplate or helmet.")
-@Examples({"set chestplate of the player to a diamond chestplate",
-		"helmet of player is neither a helmet nor air # player is wearing a block, e.g. from another plugin"})
+@Description("Equipment of living entities, i.e. the boots, leggings, chestplate or helmet.")
+@Examples({
+	"set chestplate of the player to a diamond chestplate",
+	"helmet of player is neither a helmet nor air # player is wearing a block, e.g. from another plugin"
+})
 @Since("1.0")
-public class ExprArmorSlot extends SimplePropertyExpression<LivingEntity, Slot> {
+public class ExprArmorSlot extends PropertyExpression<LivingEntity, Slot> {
+
 	static {
-		register(ExprArmorSlot.class, Slot.class, "(0¦boot[s]|0¦shoe[s]|1¦leg[ging][s]|2¦chestplate[s]|3¦helm[et][s]) [(0¦item|4¦slot)]", "livingentities");
+		register(ExprArmorSlot.class, Slot.class, "((:boots|:shoes|leggings:leg[ging]s|chestplate:chestplate[s]|helmet:helmet[s]) [(0¦item|4¦slot)]|armour:armo[u]r[s])", "livingentities");
```

Comment:
Why disallowing `boot` (singular)? this is a breaking change that is not needed.
The rest of the parse marks can use parse tags instead of mixing.

Reply:
Then I would mark it `breaking change` even tho it's a minor one but shall be noted in release notes